framework/tst/dslabs/framework/testing/newviz/DebuggerWindow.java

```diff
@@ -332,6 +353,37 @@ private void updateInvariants() {
                 label.setToolTipText(invariant.detail(currentState.state()));
             }
         }
+
+        for (Pair<StatePredicate, JLabel> e : prunes) {
+            StatePredicate prune = e.getKey();
+            JLabel label = e.getValue();
+
+            if (prune.test(currentState.state())) {
+                // Not sure what icon to put here: scissors? then what icon to use for the
+                // un-pruned?
+                label.setIcon(Utils.makeIcon(FontAwesome.LOCK,
+                        UIManager.getColor("warningColor")));
+                label.setToolTipText(prune.detail(currentState.state()));
+            } else {
+                label.setIcon(Utils.makeIcon(FontAwesome.UNLOCK));
+            }
+        }
+
+        for (Pair<StatePredicate, JLabel> e : goals) {
+            StatePredicate goal = e.getKey();
+            JLabel label = e.getValue();
+
+            if (goal.test(currentState.state())) {
+                label.setIcon(Utils.makeIcon(FontAwesome.CHECK_CIRCLE,
```

Comment:
I think we need to null out the tool tip text here? That's my bad, though. I didn't do it with the invariants.

Reply:
Fixed in https://github.com/emichael/dslabs/pull/29/commits/08217bca507baf03f3b2450682dc8a82a830403d.